tools/site_search.php

```diff
@@ -0,0 +1,354 @@
+            "url" => "$code_url/quiz/start.php",
+            "desc" => _("Interactive Quizzes and Tutorials"),
+            "alias" => "quiz",
+        ],
+        "statistics" => [
+            "url" => "$code_url/stats/stats_central.php",
+            "desc" => _("Statistics Central"),
+            "alias" => "stats",
+        ],
+        "tasks" => [
+            "url" => "$code_url/tasks.php",
+            "desc" => _("Task Center"),
+        ],
+    ];
+
+    if (get_url_for_forum()) {
+        $jump_words["forum"] = [
+            "url" => get_url_for_forum(),
+            "desc" => _("Forums"),
+        ];
+        $jump_words["inbox"] = [
+            "url" => get_url_for_inbox(),
+            "desc" => _("Inbox"),
+        ];
+    }
+
+    if (!empty($wiki_url)) {
+        $jump_words["wiki"] = [
+            "url" => $wiki_url,
+            "desc" => _("Wiki"),
+        ];
+    }
+
+    if (!empty($blog_url)) {
+        $jump_words["blog"] = [
+            "url" => $blog_url,
+            "desc" => _("Blog"),
+        ];
+    }
+
+    if (user_is_PM()) {
+        $jump_words["pm"] = [
+            "url" => "$code_url/tools/project_manager/projectmgr.php",
+            "desc" => _("Project Management"),
+        ];
+        $jump_words["rfm"] = [
+            "url" => "$code_url/tools/project_manager/remote_file_manager.php",
+            "desc" => _("Remote File Manager"),
+        ];
+    }
+
+    if (user_is_a_sitemanager()) {
+        $jump_words["sa"] = [
+            "url" => "$code_url/tools/site_admin/index.php",
+            "desc" => _("Site Administration"),
+        ];
+    }
+
+    if ($flatten_aliases) {
+        foreach ($jump_words as $key => &$data) {
+            if (isset($data["alias"])) {
+                $alias = $data["alias"];
+                unset($data["alias"]);
+                $jump_words[$alias] = $data;
+            }
+        }
+    }
+
+    ksort($jump_words);
+    return $jump_words;
+}
+
+function get_activity_jumps(): array
+{
+    global $code_url;
+
+    $activity_jumps = [];
+    foreach (Activities::get_all() as $activity) {
+        if ($activity instanceof Round) {
+            $activity_jumps[$activity->id] = "$code_url/tools/proofers/round.php?round_id=$activity->id";
+        } elseif ($activity instanceof Pool) {
+            $activity_jumps[$activity->id] = "$code_url/tools/pool.php?pool_id=$activity->id";
+        } elseif ($activity->id === "SR") {
+            // SR is such a special snowflake
+            $activity_jumps[$activity->id] = "$code_url/tools/post_proofers/smooth_reading.php";
+        }
+    }
+
+    return $activity_jumps;
+}
+
+function jump_search(string $query)
+{
+    $query = strtolower($query);
+    $jump_words = get_jump_words(true);
+    if (isset($jump_words[$query])) {
+        metarefresh(0, $jump_words[$query]["url"], "", "", true);
+    }
+}
+
+function smart_search(string $query)
+{
+    global $code_url;
+
+    // is it a valid projectID?
```

Comment:
can a user name match a project id? Same for can a user name be one of our jumps?

Reply:
Technically a username could match a project ID and in that case the requester would be taken to the project, not the user. They could use the `user:` prefix to specify they want the user not the project.

Most all of our jump words would be valid usernames. The jump words are evaluated first for this reason. See also [this line](https://github.com/DistributedProofreaders/dproofreaders/pull/1184/files#diff-05373f93c16a82c90bebb30d9ec57edeaf43a5e3d8e35aae34d1f1ad2b59ab21R18).